File: android/versioned-abis/expoview-abi39_0_0/src/main/java/abi39_0_0/host/exp/exponent/modules/api/screens/ScreenStackHeaderSubview.java
package abi39_0_0.host.exp.exponent.modules.api.screens;

import android.view.View;
import android.view.ViewParent;

import abi39_0_0.com.facebook.react.bridge.ReactContext;
import abi39_0_0.com.facebook.react.uimanager.UIManagerModule;
import abi39_0_0.com.facebook.react.views.view.ReactViewGroup;

public class ScreenStackHeaderSubview extends ReactViewGroup {

  public enum Type {
    LEFT,
    CENTER,
    RIGHT,
    BACK
  }

  private int mReactWidth, mReactHeight;

  @Override
  protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
    if (MeasureSpec.getMode(widthMeasureSpec) == MeasureSpec.EXACTLY &&
            MeasureSpec.getMode(heightMeasureSpec) == MeasureSpec.EXACTLY) {
      // dimensions provided by react
      mReactWidth = MeasureSpec.getSize(widthMeasureSpec);
      mReactHeight = MeasureSpec.getSize(heightMeasureSpec);
      ViewParent parent = getParent();
      if (parent != null) {
        forceLayout();
        ((View) parent).requestLayout();
      }
    }
    setMeasuredDimension(mReactWidth, mReactHeight);
  }

  @Override
  protected void onLayout(boolean changed, int left, int top, int right, int bottom) {
    // no-op
  }

  private Type mType = Type.RIGHT;

  public ScreenStackHeaderSubview(ReactContext context) {
    super(context);
  }

  public void setType(Type type) {
    mType = type;
  }

  public Type getType() {
    return mType;
  }
}

File: android/versioned-abis/expoview-abi44_0_0/src/main/java/abi44_0_0/host/exp/exponent/modules/api/screens/ScreenStackHeaderSubview.kt
package abi44_0_0.host.exp.exponent.modules.api.screens

import android.annotation.SuppressLint
import android.view.View
import abi44_0_0.com.facebook.react.bridge.ReactContext
import abi44_0_0.com.facebook.react.views.view.ReactViewGroup

@SuppressLint("ViewConstructor")
class ScreenStackHeaderSubview(context: ReactContext?) : ReactViewGroup(context) {
  private var mReactWidth = 0
  private var mReactHeight = 0
  var type = Type.RIGHT
  override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
    if (MeasureSpec.getMode(widthMeasureSpec) == MeasureSpec.EXACTLY &&
      MeasureSpec.getMode(heightMeasureSpec) == MeasureSpec.EXACTLY
    ) {
      // dimensions provided by react
      mReactWidth = MeasureSpec.getSize(widthMeasureSpec)
      mReactHeight = MeasureSpec.getSize(heightMeasureSpec)
      val parent = parent
      if (parent != null) {
        forceLayout()
        (parent as View).requestLayout()
      }
    }
    setMeasuredDimension(mReactWidth, mReactHeight)
  }

  override fun onLayout(changed: Boolean, left: Int, top: Int, right: Int, bottom: Int) {
    // no-op
  }

  enum class Type {
    LEFT, CENTER, RIGHT, BACK
  }
}
